### PR TITLE
Fix connection hanging on linux

### PR DESCRIPTION
**Pull request by lucaslorentz**
Our integration tests use Stubbery and they are very flaky on linux because connections hangs from time to time.

After some investigation, I realized it is much more stable if the request body is fully read by Stubbery.

### CODE REVIEW

File: src/Stubbery/StreamExtensions.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static Task<string> ReadAsStringAsync(this Stream stream)
 
             using (var sr = new StreamReader(stream))
             {
-                return sr.ReadToEndAsync();
+                return await sr.ReadToEndAsync();
```
Comment (lucaslorentz) [inline src/Stubbery/StreamExtensions.cs]: Without awaiting, the stream was disposed before finishing reading the content.

I changed the tests to reproduce the issue, then fixed it.

File: src/Stubbery/ApiStubRequestHandler.cs [hunk 2]
```diff
@@ -17,13 +18,21 @@ public ApiStubRequestHandler(ICollection<Setup> configuredEndpoints)
 
         public async Task HandleAsync(HttpContext httpContext)
         {
-            httpContext.Response.StatusCode = StatusCodes.Status404NotFound;
+            using (var bodyStream = new MemoryStream())
+            {
+                // Ensure the request body is fully read to avoid hanging connections on linux
+                await httpContext.Request.Body.CopyToAsync(bodyStream);
```
Comment (lucaslorentz) [inline src/Stubbery/ApiStubRequestHandler.cs]: This is the hacky fix for unstable connections on linux.

I tried several approaches that didn't work:
- `httpContext.Request.Body.Dispose()`
- `httpContext.Request.Body.Close()`, this one is not available on netstandard
- `httpContext.Request.Body.Seek`

File: src/Stubbery/StreamExtensions.cs (hunk 1 above)
Comment (markvincze) [reply src/Stubbery/StreamExtensions.cs]: Nice catch!

**Comment by markvincze** [comment]
Great job, thanks for the fix! 👍
Can you also bump the version number in `Stubbery.csproj` to `1.2.6`?

**Comment by lucaslorentz** [comment]
Version bumped, do you want me to squash commits?

**Pull request merged**